src/libmongoc/src/mongoc/mongoc-collection-private.h

```diff
@@ -39,6 +39,34 @@ struct _mongoc_collection_t {
    bson_t *gle;
 };
 
+struct _mongoc_index_options_t {
+   bool background;
+   char *name;
+   bool unique;
+   bson_t *partial_filter_expression;
+   bool sparse;
+   int expire_after_seconds;
+   bool hidden;
+   bson_t *storage_engine;
+   bson_t *weights;
+   char *default_language;
+   char *anguage_override;
+   int text_index_version;
+   int bits;
+   int two_d_sphere_index_version;
+   int index_min;
+   int index_max;
+   int bucket_size;
+   bson_t *collation;
+   bson_t *wildcard_projection;
+};
+
+
+struct _mongoc_index_model_t {
+   bson_t *key;
+   char *name;
+   mongoc_index_options_t *opts;
```

Comment:
Maybe this should be a `bson_t *` instead.

Reply:
I strongly suggest using a `bson_t*`, as that makes the driver more flexible going forward. Considering we don't even model most command options (e.g. [`find_with_opts`](http://mongoc.org/libmongoc/current/mongoc_collection_find_with_opts.html)) I think `bson_t*` is also more consistent with other libmongoc APIs.